sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+inherit desktop python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti cxx fortran pch +python static-libs test"
+
+REQUIRED_USE="
+	|| ( cxx python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cxx )
+	python? ( cti )
+	static-libs? ( cxx )
+	^^ ( $(python_gen_useflags 'python3*') )
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
+	dev-libs/libfmt:0=
+	dev-util/scons[${PYTHON_USEDEP}]
+	sci-libs/sundials
+	fortran? (
+		sci-libs/sundials[fortran]
+		sys-devel/gcc[fortran]
```

Comment:
Doesn't it work with other fortran compilers? If it does, `virtual/fortran` is probably more appropriate.

Reply:
The `virtual/fortran` package refers for next compilers:
```
sys-devel/gcc[fortran,openmp?]
sys-devel/gcc-apple[fortran,openmp?]
dev-lang/ekopath
dev-lang/ifc
```
while `dev-lang/ekopath` and `dev-lang/ifc` aren't presented in portage tree.

Also (by cantera developers):
> The 2017 and newer versions of the Intel compiler are NOT RECOMMENDED because of a bug in the C compiler, which has been reported to Intel: [https://software.intel.com/en-us/forums/intel-c-compiler/topic/684987](https://software.intel.com/en-us/forums/intel-c-compiler/topic/684987)

As for `sys-devel/gcc-apple`: I can't test it because of it's `KEYWORDS="~ppc-macos ~x64-macos ~x86-macos"` that ARCHs aren't available for me. Is it worth to add `virtual/fortran` in this case anyway?

P.S. I don't have any information about availability to build package with PGI Group C++ and Fortran compilers. The last also isn't listed within `virtual/fortran`.

Reply:
If the package uses standards-compliant Fortran, there's no reason why virtual wouldn't work.

Reply:
Also, don't forget that depending on gcc won't force it anyway. So you'd just be introducing a fake dependency.

That said, I think you ought to use fortran-2 eclass when you're building some Fortran stuff. Note that it exports pkg_setup, so you'd have to have your own `pkg_setup()` calling appropriate eclasses, with appropriate conditions if applicable.

Reply:
I added 
```
FORTRAN_NEEDED=fortran
FORTRAN_STANDARD=90
```
before `inherit` and `fortran-2` to `inherit` with removing block 
```
	fortran? (
		sci-libs/sundials[fortran]
		sys-devel/gcc[fortran]
	)
```
and `${EPYTHON}` now returns empty string instead of `python3.6`.
If I remove the `fortran-2` eclass and approprote block of it before inherit then `${EPYTHON}` set up to `python3.6` again.

If I place previously removed `python_setup` calling in `if use cti ; then` block then `${EPYTHON}` also set up to `python3.6` again.

Reply:
Yes, since you need Python for SCons.

Reply:
You'd probably need to call fortran from there too.

Reply:
Done.
Without addition of `fortran-2_pkg_setup` into `pkg_setup()` section.

Reply:
I assume I also must pass the `tc-getFC` from `toolchain-funcs` to scons build options to specify Fortran90 compiler explicitly instead of it's autodetection by build system.

Reply:
Done: update with using of `$(tc-getFC)` function